src/components/connections/Connection.ts

```diff
@@ -0,0 +1,177 @@
+import * as vscode from "vscode";
+import { Engine as QueryEngine } from "../Query";
+import { INode } from "./INode";
+import { Status } from "./Status";
+import { ConnectionNode, InfluxDBConectionsKey } from "./ConnectionNode";
+
+const uuidv1 = require("uuid/v1");
+export interface InfluxDBConnection {
+  readonly name: string;
+  readonly hostNport: string;
+  readonly token: string;
+  readonly org: string;
+}
+
+export class InfluxDBTreeDataProvider
+  implements vscode.TreeDataProvider<INode> {
+  public _onDidChangeTreeData: vscode.EventEmitter<
+    INode
+  > = new vscode.EventEmitter<INode>();
+
+  public readonly onDidChangeTreeData: vscode.Event<INode> = this
+    ._onDidChangeTreeData.event;
+
+  constructor(
+    private context: vscode.ExtensionContext,
+    private outputChannel: vscode.OutputChannel
+  ) {}
+
+  getTreeItem(element: INode): vscode.TreeItem | Thenable<vscode.TreeItem> {
+    return element.getTreeItem(this.context);
+  }
+
+  getChildren(element?: INode): Thenable<INode[]> | INode[] {
+    if (!element) {
+      return this.getConnectionNodes(this.outputChannel);
+    }
+    return element.getChildren(this.outputChannel);
+  }
+
+  public refresh(element?: INode): void {
+    this._onDidChangeTreeData.fire(element);
+  }
+
+  public async addConnection() {
+    const name = await vscode.window.showInputBox({
+      prompt: "Give a name of the influxdb connection",
+      placeHolder: "name",
+      ignoreFocusOut: true
+    });
+    if (!name) {
+      return;
```

Comment:
should we show something to the user saying this field is required if they leave it blank?

Reply:
i copied from vscode mysql plugin, so if the user leaves it as blank, we could assume they gave up.